src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Tests/INullArrayTests.cs

```diff
@@ -0,0 +1,244 @@
+            int[] array = new int[5];
+
+            testInterface.SingleNullArrayWithLength(5, array);
+
+            Assert.Equal(new int[] { 0, 2, 4, 6, 8 }, array);
+        }
+
+        [Fact]
+        public void MultipleArrays_SomeNull_DoesNotCrash()
+        {
+            var testInterface = CreateTestInterface();
+            int[] array1 = new int[3];
+            int[] array3 = new int[3];
+
+            testInterface.MultipleArraysSharedLength(3, array1, null, array3);
+
+            Assert.Equal(new int[] { 0, 1, 2 }, array1);
+            Assert.Equal(new int[] { 0, 100, 200 }, array3);
+        }
+
+        [Fact]
+        public void MultipleArrays_AllNull_DoesNotCrash()
+        {
+            var testInterface = CreateTestInterface();
+
+            testInterface.MultipleArraysSharedLength(5, null, null, null);
+        }
+
+        [Fact]
+        public void NonBlittableArray_Null_DoesNotCrash()
+        {
+            var testInterface = CreateTestInterface();
+
+            testInterface.NonBlittableNullArray(10, null);
+        }
+
+        [Fact]
+        public void NonBlittableArray_ValidArray_WorksNormally()
+        {
+            var testInterface = CreateTestInterface();
+            var array = new IntStructWrapper[3];
+
+            testInterface.NonBlittableNullArray(3, array);
+
+            Assert.Equal(0, array[0].Value);
+            Assert.Equal(3, array[1].Value);
+            Assert.Equal(6, array[2].Value);
+        }
+
+        [Fact]
+        public void ZeroLength_NullArray_DoesNotCrash()
+        {
+            var testInterface = CreateTestInterface();
+
+            testInterface.ZeroLengthArray(0, null);
+        }
+
+        [Fact]
+        public void LargeLength_NullArray_DoesNotCrash()
+        {
+            var testInterface = CreateTestInterface();
+
+            testInterface.LargeLengthNullArray(int.MaxValue, null);
+        }
+
+        [Fact]
+        public void SpanNull_DoesNotCrash()
+        {
+            var testInterface = CreateTestInterface();
+
+            var (__this, __vtable) = ((IUnmanagedVirtualMethodTableProvider)testInterface).GetVirtualMethodTableInfoForKey(typeof(INullArrayCases));
+            int length = 10;
+            var __target = (delegate* unmanaged[MemberFunction]<void*, int, int**, int>)__vtable[11];
+            int* __span_native = null;
+            int __invokeRetVal = __target(__this, length, &__span_native);
+            Marshal.ThrowExceptionForHR(__invokeRetVal);
+        }
+
+        [Fact]
+        public void SpanNull_WithZeroLength_DoesNotCrash()
+        {
+            var testInterface = CreateTestInterface();
+
+            var (__this, __vtable) = ((IUnmanagedVirtualMethodTableProvider)testInterface).GetVirtualMethodTableInfoForKey(typeof(INullArrayCases));
+            int length = 0;
+            var __target = (delegate* unmanaged[MemberFunction]<void*, int, int**, int>)__vtable[11];
```

Comment:
Indexing into the vtable is fragile. Is this a pattern we use elsewhere? I would have assumed we wrap the CCW in an RCW and verify it that way.

Reply:
I thought the marshaller would marshal `Span` to a non-null value, so I did this to make sure the parameter would be passed as null. It looks like the marshaller will actually pass a null value for `default`, so it can use an RCW instead. I'll update the test.